docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,277 @@
+at this time and will assume whatever budget was set aside for `docker` was included in
+either `--kube-reserved` or `--system-reserved` reservations.
+
+**Rocket**
+
+Rocket has no client/server daemon, and therefore has no explicit requirements on container-runtime
+reservation.
+
+### Node allocable
+
+The proposal makes no changes to the definition.
+
+The node will report a set of allocable compute resources defined as follows:
+
+`[Allocatable] = [Node Capacity] - [Kube-Reserved] - [System-Reserved]`
+
+### OOM Score Adjustment
+
+The `kubelet` at bootstrapping will set the `oom_score_adj` value for Kubernetes
+daemons, and any dependent container-runtime daemons.
+
+If `container-runtime` is set to `docker`, then set its `oom_score_adj=-900`
+
+### Linux Kernel Parameters
+
+The `kubelet` will set the following:
+
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w vm.panic_on_oom=0`
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w kernel/panic=1`
+* `sysctl -w kernel/panic_on_oops=10`
+
+## kubelet cgroup-root behavior under systemd
+
+The `kubelet` supports a `cgroup-root` flag which is the optional root `cgroup` to use for pods.
+
+This flag should be treated as a pass-through to the underlying configured container runtime.
+
+For example, if the container runtime is `docker` and its using the `systemd` cgroup driver, then
+it should take the form of a `systemd` slice.  For example, `--cgroup-root=foo-bar.slice` would parent
+all of the pod's container under the `foo-bar.slice` part of the hierarchy.
+
+## kubelet accounting for end-user pods
+
+This proposal re-enforces that it is inappropriate at this time to depend on `--cgroup-root` as the
+primary mechanism to distinguish and account for end-user pod compute resource usage.
+
+Instead, the `kubelet` can and should sum the usage of each running `pod` on the node to account for
+end-user pod usage separate from system-reserved and kubernetes-reserved accounting via `cAdvisor`.
+
+## Known issues
+
+### Docker runtime support for --cgroup-parent
+Docker versions <= 1.0.9 did not have proper support for `-cgroup-parent` flag on `systemd`.  This
+was fixed in this PR (https://github.com/docker/docker/pull/18612).  As result, it's expected
+that containers launched by the `docker` daemon may continue to go in the default `system.slice` and
+appear to be counted under system-reserved node usage accounting.
+
+If operators run with later versions of `docker`, they can avoid this issue via the use of `cgroup-root`
+flag on the `kubelet`, but this proposal makes no requirement on operators to do that at this time, and
+this can be revisited if/when the project adopts docker 1.10.
+
+### Ability to distinguish user-reserved from system-reserved
+
+It is impractical to assume that all operators will never have to SSH into machines to debug their
+own agents on the system.  On `systemd` environments, user sessions are tracked in `user.slice`.
```

Comment:
> On `systemd` environments, user sessions are tracked in `user.slice`.

Strictly speaking, this is a necessary but not sufficient condition; a system must also be using `pam_systemd`, as systemd's concept of users is intimately tied in to that (see e.g. discussion [here](https://github.com/coreos/go-systemd/pull/87#issuecomment-104466255)). As just one example, currently CoreOS Linux has no `user.slice`.